packages/esbuild/esbuild.bzl

```diff
@@ -0,0 +1,220 @@
+        execution_requirements = {
+            "no-remote-exec": "1",
+        },
+    )
+
+    return [
+        DefaultInfo(files = depset(outputs + [jsconfig_file])),
+    ]
+
+esbuild_bundle = rule(
+    attrs = {
+        "define": attr.string_list(
+            default = [],
+            doc = """A list of global identifier replacements.
+Example:
+```python
+esbuild(
+    name = "bundle",
+    define = [
+        "process.env.NODE_ENV=\\"production\\""
+    ],
+)
+```
+            """,
+        ),
+        "deps": attr.label_list(
+            default = [],
+            aspects = [module_mappings_aspect, node_modules_aspect],
+            doc = "A list of direct dependencies that are required to build the bundle",
+        ),
+        "entry_point": attr.label(
+            mandatory = True,
+            allow_single_file = True,
+            doc = "The bundle's entry point (e.g. your main.js or app.js or index.js)",
+        ),
+        "external": attr.string_list(
+            default = [],
+            doc = "A list of module names that are treated as external and not included in the resulting bundle",
+        ),
+        "format": attr.string(
+            values = ["iife", "cjs", "esm", ""],
+            mandatory = False,
+            doc = """The output format of the bundle, defaults to iife when platform is browser
+and cjs when platform is node. If performing code splitting, defaults to esm""",
+        ),
+        "minify": attr.bool(
+            default = False,
+            doc = """Minifies the bundle with the built in minification.
+Removes whitespace, shortens identifieres and uses equivalent but shorter syntax.
+
+Sets all --minify-* flags
+            """,
+        ),
+        "output": attr.output(
+            mandatory = False,
+            doc = "Name of the output file when bundling",
+        ),
+        "output_map": attr.output(
+            mandatory = False,
+            doc = "Name of the output source map when bundling",
+        ),
+        "platform": attr.string(
+            default = "browser",
+            values = ["node", "browser", "neutral", ""],
+            doc = "The platform to bundle for",
+        ),
+        "sources_content": attr.bool(
+            mandatory = False,
+            default = False,
+            doc = "If False, omits the `sourcesContent` field from generated source maps",
+        ),
+        "splitting": attr.bool(
+            default = False,
+            doc = """If true, esbuild produces an output directory containing all the output files from code splitting 
+            """,
+        ),
+        "srcs": attr.label_list(
+            allow_files = True,
+            default = [],
+            doc = """Non-entry point JavaScript source files from the workspace.
+
+You must not repeat file(s) passed to entry_point""",
+        ),
+        "target": attr.string(
+            default = "es2015",
```

Comment:
feels like we're missing an `args` to let you pass arbitrary other arguments to esbuild

and once we have that, we should defend why Bazel needs to know about all of them. I think we only need to know the ones that are needed to predict output paths?